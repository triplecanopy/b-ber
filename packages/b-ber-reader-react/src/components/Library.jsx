import React from 'react'

const Library = props => (
  <div>
    {props.books.map((book, i) => (
      <button
        key={i}
        onClick={() => props.handleClick(book)}
        style={{
          width: 200,
          height: 300,
          display: 'inline-block',
          backgroundImage: `url(${book.cover})`,
          backgroundSize: 'contain',
          backgroundRepeat: 'no-repeat',
          backgroundPosition: 'center',
        }}
      >
        {book.name}
      </button>
    ))}
  </div>
)

export default Library
