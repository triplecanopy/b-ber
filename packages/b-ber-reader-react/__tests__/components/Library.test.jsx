/* eslint-disable react/jsx-props-no-spreading */
/* global jsdom */
/* eslint-disable prefer-destructuring */

import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import Library from '../../src/components/Library'
import history from '../../src/lib/History'

describe('Library', () => {
  test('renders the component', () => {
    jsdom.reconfigure({ url: 'http://localhost:3000/' })

    const props = {
      books: [{ title: 'Book 1', cover: 'cover-1.jpg' }],
      handleClick: ({ title, url }) => history.push(title, { bookURL: url }),
    }

    const component = render(<Library {...props} />)
    const { container } = component

    expect(container).toMatchSnapshot()

    const button = container.querySelector('button')
    expect(button.style.backgroundImage).toMatch(/^url\(cover-1.jpg\)/)

    fireEvent.click(button)
    expect(window.location.href).toBe('http://localhost:3000/Book%201')
  })
})
