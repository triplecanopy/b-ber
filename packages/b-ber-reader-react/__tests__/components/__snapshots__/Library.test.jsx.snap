// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library renders the component 1`] = `
<div>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(cover-1.jpg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "display": "inline-block",
        "height": 300,
        "width": 200,
      }
    }
  />
</div>
`;
